reporting/CassandraBackend.cpp

```diff
@@ -573,49 +573,87 @@ CassandraBackend::fetchBookOffers(
     std::optional<ripple::uint256> const& cursor) const
 {
     CassandraStatement statement{selectBook_};
-    statement.bindBytes(book);
-    auto index = getIndexOfSeq(sequence);
-    if (!index)
-        return {};
-    BOOST_LOG_TRIVIAL(info) << __func__ << " index = " << std::to_string(*index)
-                            << " book = " << ripple::strHex(book);
-    statement.bindInt(*index);
-    if (cursor)
-        statement.bindBytes(*cursor);
-    else
-    {
-        ripple::uint256 zero = {};
-        statement.bindBytes(zero);
-    }
-    statement.bindUInt(limit);
-    CassandraResult result = executeSyncRead(statement);
 
-    BOOST_LOG_TRIVIAL(debug) << __func__ << " - got keys";
+    auto rng = fetchLedgerRange();
+    
+    if(!rng)
+        return {{},{}};
+
     std::vector<ripple::uint256> keys;
-    if (!result)
-        return {{}, {}};
-    do
+    uint32_t upper = sequence;
+    auto lastPage = rng->maxSequence - (rng->maxSequence % 256);
+
+    if (sequence != rng->minSequence)
     {
-        keys.push_back(result.getUInt256());
-    } while (result.nextRow());
+        upper = (sequence >> 8) << 8;
+        if (upper != sequence)
+            upper += (1 << 8);
+    
+        statement.bindBytes(book.data(), 24);
+        statement.bindInt(upper);
+
+        BOOST_LOG_TRIVIAL(info) << __func__ << " upper = " << std::to_string(upper)
+                                << " book = " << ripple::strHex(std::string((char*)book.data(), 24));
+
+        if (cursor)
+        {
+            auto object = fetchLedgerObject(*cursor, sequence);
+
+            if(!object)
+                return {{}, {}};
+
+            ripple::SerialIter it{object->data(), object->size()};
+            ripple::SLE offer{it, *cursor};
+            ripple::uint256 bookDir = offer.getFieldH256(ripple::sfBookDirectory);
+
+            statement.bindBytes(bookDir.data() + 24, 8);
+            statement.bindBytes(*cursor);
+        }
+        else
+        {
+            ripple::uint256 zero = beast::zero;
+            statement.bindBytes(zero.data(), 8);
+            statement.bindBytes(zero);
+        }
+
+        // statement.bindUInt(limit);
+        CassandraResult result = executeSyncRead(statement);
+
+        BOOST_LOG_TRIVIAL(debug) << __func__ << " - got keys";
+        if (!result)
+        {
+            std::cout << "could not sync read" << std::endl;
```

Comment:
Let's change these `std::cout` print statements to log messages using `BOOST_LOG_TRIVIAL`. You can add them at whatever log level you think is appropriate.